inc/class-provider.php

```diff
@@ -35,6 +35,55 @@ public function __construct( Factory $factory ) {
 		$this->factory = $factory;
 	}
 
+	/**
+	 * Return the provider ID.
+	 *
+	 * @return string
+	 */
+	public function get_id(): string {
+
+		return 'wordpress'; // phpcs:ignore
+	}
+
+	/**
+	 * Return the provider name.
+	 *
+	 * @return string
+	 * @throws Exception If site name cannot be found
+	 */
+	public function get_name(): string {
```

Comment:
Hm, instead of this rather complicated and performance-heavy approach, couldn't we make the name be _External WordPress Media_, and filter it? That way you can adapt as necessary and we don't need an extra request (that may fail). For a project this is in use, the name would actually have to be _Shared Media_.

If anyone wants to, they can use the filter and hook up something like the code that you have provided here. Or, because they know what the name of the site is anyway, they just use the filter and pass the literal name.

Reply:
Hmm, it is an argument for allowing the name to be provided during registration potentially too as it'd be easier to filter in a unified way 🤔  At that point might as well provide the ID and name during registration then these functions don't need to be abstracts on the provider, they could look up values from the registry object directly.

I'm not against just providing a filter at any rate, just thought a bit of automation might help and true - for setups not using a persistent object cache this implementation is no good, should've been a transient. I'll update.

Reply:
It makes sense to me that each provider implementation comes with a unique identifier. If we keep that separate from the object, we have to pass multiple values here and there.
The name, too, would ideally be specified by the provider. I was only thinking about the current code because, if you want to override the name using the filter, you would still ask for the remote site's name first (if cache is empty, yes), and then scrap it. So I thought that providing a default, yet filterable name would be good (enough).

Reply:
Reverted to a filter, I can mess with the Altis implementation name in the media module code.